app/Domain/Timesheets/Templates/showMy.tpl.php

```diff
@@ -263,69 +273,44 @@
                 </thead>
                 <tbody>
                     <?php
+                        $colSum = [];
                         // @todo: move all this calculations into the service the timesheets class.
-                        $rowKeys = [
-                           "hoursMonday",
-                           "hoursTuesday",
-                           "hoursWednesday",
-                           "hoursThursday",
-                           "hoursFriday",
-                           "hoursSaturday",
-                           "hoursSunday",
-                        ];
-                        $totalHours = [];
-                        foreach ($rowKeys as $key) {
-                            $totalHours[$key] = 0;
-                        }
-
                         foreach ($tpl->get('allTimesheets') as $timeRow) {
-                            // Calculate totals.
-                            $rowSum = 0;
-                            foreach ($rowKeys as $key) {
-                                $totalHours[$key] += $timeRow[$key];
-                                $rowSum += $timeRow[$key];
-                            }
 
                             $timesheetId = "new";
-
-
-                            $workDatesArray = [];
-                            foreach (explode(",", $timeRow["workDates"]) as $workDate) {
-                                $date = new \Carbon\Carbon($workDate, 'UTC');
-                                $workDatesArray[format($date, 'short')] = $date;
-                             }
-
-                            /** @var \Carbon\Carbon $currentDate */
-                            $currentDate = $tpl->get("dateFrom")->copy();
                             ?>
 
-                        <tr class="gradeA timesheetRow">
+                            <tr class="gradeA timesheetRow">
                             <td width="14%"><?php $tpl->e($timeRow["clientName"]); ?> // <?php $tpl->e($timeRow["name"]); ?></td>
                             <td width="14%"><?php $tpl->e($timeRow["headline"]); ?></td>
                             <td width="10%"><?php echo $tpl->__($tpl->get('kind')[$timeRow['kind']]); ?></td>
 
-                            <?php foreach ($rowKeys as $i => $key) { ?>
-                                <td width="7%" class="row<?php echo $days[$i]; ?><?php if ($currentDate->isToday()) {
+                            <?php for ($i = 1; $i < 8; $i++) {
```

Comment:
Maybe a comment about these "for" loop that start at 1 and runs to 8? Would is be possible to make a foreach loop over the data (foreach are consider more safe)

Reply:
Thanks for the call out. I refactored that.